dm/ctl/master/batch_task.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package master
+
+import (
+	"context"
+	"errors"
+	"os"
+	"sort"
+	"sync"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pingcap/dm/dm/ctl/common"
+	"github.com/pingcap/dm/dm/pb"
+)
+
+const batchSizeFlag = "batch-size"
+const defaultBatchSize = 5
+
+type batchTaskResult struct {
+	Result       bool                 `json:"result"`
+	Msg          string               `json:"msg"`
+	BatchResults []*operateTaskResult `json:"batchResults"`
+}
+
+type operateTaskResult struct {
+	TaskName string                     `json:"taskName"`
+	Op       string                     `json:"op"`
+	Result   bool                       `json:"result"`
+	Msg      string                     `json:"msg"`
+	Sources  []*pb.CommonWorkerResponse `json:"sources"`
+}
+
+// NewBatchTaskCmd create a new BatchTask command.
+func NewBatchTaskCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "batch-task <command>",
+		Short: `Operate subtasks bound to an upstream MySQL/MariaDB source in batch mode`,
+	}
+
+	// control workload to dm-cluster for sources with large number of tasks.
+	cmd.PersistentFlags().Int(batchSizeFlag, defaultBatchSize, "batch size")
+
+	cmd.AddCommand(
+		&cobra.Command{
+			Use:   "pause <source-name>",
+			Short: "pause subtasks bound to an upstream MySQL/MariaDB source in batch mode",
+			RunE:  batchPauseTaskFunc,
+		},
+		&cobra.Command{
```

Comment:
maybe we should also provide batch stop. A paused task still forbids its source to be removed

Reply:
which scenario needs to stop tasks in batch mode?

Reply:
a prerequisite of `operate-source stop` is no task is using the source.